lib/pure/hashes.nim

```diff
@@ -254,13 +251,14 @@ proc hashIgnoreCase*(sBuf: string, sPos, ePos: int): Hash =
     var a = "ABracadabRA"
     doAssert hashIgnoreCase(a, 0, 3) == hashIgnoreCase(a, 7, 10)
 
-  var h: Hash = 0
-  for i in sPos..ePos:
-    var c = sBuf[i]
-    if c in {'A'..'Z'}:
-      c = chr(ord(c) + (ord('a') - ord('A'))) # toLower()
-    h = h !& ord(c)
-  result = !$h
+  var
+    remainingLength = ePos - sPos + 1
+    lowerString = newStringOfCap(remainingLength)
+    i = sPos
+  while i <= ePos:
+    lowerString.addLowercaseChar(sBuf[i])
```

Comment:
If string is empty and we pass sPos = ePos = 0, this probably crash.
Test case as well.

(Original code probably suffer from the same issue).